git_machete/client.py

```diff
@@ -1674,16 +1674,27 @@ def checkout_github_prs(self, pr_no: int) -> None:
         repo: str
         remote: str
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        current_user: Optional[str] = git_machete.github.derive_current_user_login()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
         print(f"Fetching {remote}...")
         self.__git.fetch_remote(remote)
-        self.flush_caches()
 
         pr = get_pull_request_by_number_or_none(pr_no, org, repo)
         if not pr:
             raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
-        if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
-            raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote}`.")
+        if pr.full_repository_name:
+            if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
+                remote_from_pr: str = pr.full_repository_name.split('/')[0]
+                if remote_from_pr not in self.__git.get_remotes():
+                    self.__git.add_remote(remote_from_pr, pr.repository_url)
+                print(f"Fetching {remote_from_pr}...")
+                self.__git.fetch_remote(remote_from_pr)
+                if '/'.join([remote_from_pr, pr.head]) not in self.__git.get_remote_branches():
+                    raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote_from_pr}`.")
+        else:
+            warn(f'Pull request #{pr_no} comes from fork and its repository is already deleted. Read only view of this PR will be checked out.')
```

Comment:
Okay, I didn't know this was the problem with #241 😅

One remark to `Read only view` in the error message: in a sense, even when the original repository still exists... you still kind-of get a read-only (or rather, fetch-only) view of that branch &mdash; unless you have write permission on the original repository.

Reply:
So! I'd rather say sth like
```suggestion
            warn(f'Pull request #{pr_no} comes from fork and its repository is already deleted. No remote tracking data will be set up for `{pr.head}` branch.')
```